packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -1791,7 +1791,13 @@ EmberRouter.reopen(Evented, {
    @private
  */
   url: computed(function(this: Router<Route>) {
-    return get(this, 'location').getURL();
+    let location = get(this, 'location');
+
+    if (typeof location !== 'string') {
```

Comment:
Could avoid the negation by swapping the returns

Reply:
That's a good idea.  I'll do that.